plugins/media/wowza/services/LiveConversionProfileService.php

```diff
@@ -257,39 +366,31 @@ protected function appendLiveParams(LiveStreamEntry $entry, WowzaMediaServerNode
 		$video->addChild('Transcoder', $mediaServer ? $mediaServer->getTranscoder() : WowzaMediaServerNode::DEFAULT_TRANSCODER);
 		$video->addChild('GPUID', $mediaServer ? $mediaServer->getGPUID() : WowzaMediaServerNode::DEFAULT_GPUID);
 		$frameSize = $video->addChild('FrameSize');
-	
-		if(!$liveParams->getWidth() && !$liveParams->getHeight())
-		{
-			$frameSize->addChild('FitMode', 'match-source');
-		}
-		elseif($liveParams->getWidth() && $liveParams->getHeight())
-		{
-			$frameSize->addChild('FitMode', 'fit-height');
-			$frameSize->addChild('Width', $liveParams->getWidth());
-			$frameSize->addChild('Height', $liveParams->getHeight());
-		}
-		elseif($liveParams->getWidth())
+		
+		$frameSize->addChild('FitMode', $flavorResolutionInfo['fitMode']);
+		if (isset($flavorResolutionInfo['width']))
 		{
-			$frameSize->addChild('FitMode', 'fit-width');
-			$frameSize->addChild('Width', $liveParams->getWidth());
+			$frameSize->addChild('Width', $flavorResolutionInfo['width']);
 		}
-		elseif($liveParams->getHeight())
+		if (isset($flavorResolutionInfo['height']))
 		{
-			$frameSize->addChild('FitMode', 'fit-height');
-			$frameSize->addChild('Height', $liveParams->getHeight());
+			$frameSize->addChild('Height', $flavorResolutionInfo['height']);
 		}
 		
 		$video->addChild('Codec', $videoCodec);
 		$video->addChild('Profile', $profile);
-		$video->addChild('Bitrate', $liveParams->getVideoBitrate() ? $liveParams->getVideoBitrate() * 1024 : 240000);
+		$video->addChild('Bitrate', $flavorBitrateValue);
 		$keyFrameInterval = $video->addChild('KeyFrameInterval');
 		$keyFrameInterval->addChild('FollowSource', 'true');
 		$keyFrameInterval->addChild('Interval', 60);
 		
-		if ($conversionExtraParam && $conversionExtraParam->skipFrameCount)
+		$skipFrameCountByMaxFramerate = $this->checkMaxFramerate($streamParametersArray['framerate'], $liveParams->getMaxFrameRate());
```

Comment:
when looking at:
if ($configuredSkipFrameRate || $skipFrameCountByMaxFramerate

you don't really need to calculate both values for "skipFrameCountByMaxFramerate" & "configuredSkipFrameRate" check the second only f the first is not defined.